tarorpc/taro.proto

```diff
@@ -508,8 +514,24 @@ message ProofFile {
     string genesis_point = 2;
 }
 
-message ProofVerifyResponse {
+message VerifyProofResponse {
     bool valid = 1;
+    DecodeProofResponse decoded_proof = 2;
+}
+
+message DecodeProofRequest {
+    bytes raw_proof = 1;
+    int64 proof_index = 2;
+}
+
+message DecodeProofResponse {
```

Comment:
Can potentially expose more data here, but I think this is good as an initial set.

Reply:
nit: split this in two different messages

```
message DecodedProof {
// decoded proof fields
}

message DecodeProofResponse {
    DecodedProof decoded_proof = 1
}

// VerifyProofResponse does not use anything called "response" then, that is supposed to be for method
// responses
message VerifyProofResponse {
    bool valid = 1;
    DecodedProof decoded_proof = 2;
}
```

Reply:
After reviewing the implementation I think we should also add `number_of_proofs` to the `DecodeProofResponse`. By default I want to get info about the last one, but I would also want to know how many are there in case I want to use the index parameter.